GVFS/GVFS.Common/Cleanup/PostFetchCleanupStep.cs

```diff
@@ -0,0 +1,47 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GVFS.Common.Cleanup
+{
+    public class PostFetchCleanupStep : GitCleanupStep
+    {
+        private const string CommitGraphLock = "commit-graph.lock";
+        private const string MultiPackIndexLock = "multi-pack-index.lock";
+        private List<string> packIndexes;
+
+        public PostFetchCleanupStep(GVFSContext context, GitObjects gitObjects, List<string> packIndexes)
+            : base(context, gitObjects)
+        {
+            this.packIndexes = packIndexes;
+        }
+
+        public override string TelemetryKey => "PostFetchCleanupStep";
+
+        protected override void RunGitAction()
+        {
+            using (ITracer activity = this.Context.Tracer.StartActivity("TryWriteMultiPackIndex", EventLevel.Informational, Keywords.Telemetry, metadata: null))
+            {
+                string multiPackIndexLockPath = Path.Combine(this.Context.Enlistment.GitPackRoot, MultiPackIndexLock);
+                this.Context.FileSystem.TryDeleteFile(multiPackIndexLockPath);
+
+                this.RunGitCommand((process) => process.WriteMultiPackIndex(this.Context.Enlistment.GitObjectsRoot));
+            }
+
+            if (this.packIndexes == null || this.packIndexes.Count == 0)

```

Comment:
Why do we skip the commit graph but always write the multi pack index?

Reply:
The `multi-pack-index` call will quickly no-op if there are no new pack-files, but there could also be new pack-files (from a `gvfs prefecth --files` call) so there may be something to do.

The commit-graph command literally cannot do anything without a list of pack-files, as it uses that to determine commits to add to the file.